File: Python/programmers/etc/level1/폰켓몬.py
def solution(nums):
    a, b = len(nums)//2, len(set(nums))
    return a if a < b else b 


nums = [3,1,2,3]
print(solution(nums)) # 2

nums = [3,3,3,2,2,4]
print(solution(nums)) # 3

nums = [3,3,3,2,2,2]
print(solution(nums)) # 2

File: Java/programmers/etc/level1/폰켓몬.java
package CodeTest.Java.programmers.etc.level1;

import java.util.*;

public class 폰켓몬 {
    public int solution(int[] nums) {
        HashSet<Integer> hs = new HashSet<>();
        for (int num : nums) hs.add(num);
        return Math.min(hs.size(), nums.length/2);
    }
}
